### ISSUE TITLE
Failing matrix integration tests with eth-tester

### ISSUE THREAD

**Issue by jomuel** [raiden-network/raiden#1616]
Some integration tests fail when run with the options `--transport=matrix --blockchain-type=tester`. This is probably caused by further concurrency issues in `MatrixTransport`. Examine them next as soon as #1525 is merged.

**Comment by andrevmatos** [comment]
For completude, what we've found so far about this issue:

Matrix tests sometimes fails and there's no clear information about why.

To debug it, we added changes both to the synapse build and to the smoketest and pytest arguments to allow to run the tests in a self-running synapse server, with `--local-matrix=none`, as well as enabling the webclient to allow peaking in the rooms between test clients.

Sometimes a client fails completely to listen to events on a room it participates in, resulting in messages being sent by the sending party but not being handled by the receiving party. Code was added to ensure the listeners were properly added to the peers rooms, which fixed some errors, but not all. @pcppcp discovered that the `requests` session connection pool was being filled after long running rest sessions, because matrix http connections are long-polling connections which doesn't get closed immediately. He added a fix to increase the connection poll to 256, which seems to get rid of this problem, although we may look into ensuring proper socket close in the future.

Some of these errors have triggered not previously tested codepath, which showed hidden bugs in refund of mediated transfers #1627 #1653 .
